core/src/main/java/org/apache/iceberg/deletes/SortingPositionOnlyDeleteWriter.java

```diff
@@ -93,9 +135,9 @@ private DeleteWriteResult writeDeletes() throws IOException {
     return writer.result();
   }
 
-  private List<CharSequence> sortedPaths() {
-    List<CharSequence> paths = Lists.newArrayList(positionsByPath.keySet());
-    paths.sort(Comparators.charSequences());
-    return paths;
+  private Iterable<CharSequence> sort(Iterable<CharSequence> paths) {
```

Comment:
What about using `Collection` for the incoming so you can test its size? Then you could check whether the size is 1 and avoid copying the list.